lha: init at 2021-01-07

svanderburg opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Adds lha: an archiver commonly used on AmigaOS, and MS-DOS/Windows. It is still popular in Japan on the latter two OSes.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

svanderburg committed:
lha: init at 2021-07-01

SuperSandro2000 commented on pkgs/tools/archivers/lha/default.nix:
```diff
@@ -0,0 +1,28 @@
+{stdenv, lib, fetchFromGitHub, autoreconfHook}:
+
+stdenv.mkDerivation {
+  name = "lha-1.14i";
```
```suggestion
  pname = "lha";
  version = "1.14i";
```

SuperSandro2000 commented on pkgs/tools/archivers/lha/default.nix:
```diff
@@ -0,0 +1,28 @@
+  src = fetchFromGitHub {
+    owner = "jca02266";
+    repo = "lha";
+    rev = "03475355bc6311f7f816ea9a88fb34a0029d975b";
```
I don't think the program is versioned very well. We should use the unstable-XXXX-XX-XX format instead.

SuperSandro2000 commented on pkgs/tools/archivers/lha/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+  meta = with lib; {
+    description = "LHa is an archiver and compressor using the LZSS and Huffman encoding compression algorithms";
+    longDescription = "LHa is an archiver and compressor commonly on AmigaOS, and MS-DOS and Windows (mostly in Japan)";
```
```suggestion
```
same text and longDescription is optional.

SuperSandro2000 commented on pkgs/tools/archivers/lha/default.nix:
```diff
@@ -0,0 +1,28 @@
+    description = "LHa is an archiver and compressor using the LZSS and Huffman encoding compression algorithms";
+    longDescription = "LHa is an archiver and compressor commonly on AmigaOS, and MS-DOS and Windows (mostly in Japan)";
+    platforms = platforms.unix;
+    maintainers = [ maintainers.sander ];
+
+    # Some of the original LhA code has been rewritten and the current author
```
```suggestion
    maintainers = [ maintainers.sander ];
    # Some of the original LhA code has been rewritten and the current author
```

svanderburg commented:
I've addressed the comments. It turns out that some Linux distributions that have this package in their repositories still use 1.14i for versioning even though they're not using an official release tarball. 

Perhaps it's still better not to deviate from our usual packaging practices. I've changed the version identifier to unstable.

svanderburg commented:
If there are no further comments, I'll merge this tonight
